crates/rust-analyzer/src/handlers.rs

```diff
@@ -947,6 +948,52 @@ pub(crate) fn handle_code_lens(
                 }),
         );
     }
+
+    if snap.config.lens.references() {
+        let ref_lenses = snap
+            .analysis
+            .file_structure(file_id)?
+            .into_iter()
+            .filter(|it| match it.kind {
```

Comment:
`it == SyntaxKind::FN`.

But it also feels like something that should be implemented in the `ide` rather than handlers. 

Something like

```rust
impl Analysis {
  fn declaration_references(&self, file_id: FileId) -> Vec<RangeInfo<usize>>
}
```